pack.sh

```diff
@@ -38,16 +38,24 @@ function lerna_scopes() {
 
 scripts/run-rosetta.sh --infuse --pkgs-from $TMPDIR/jsii.txt
 
+# Execute any pre-package steps for the jsii modules here:
```

Comment:
I'm not very familiar with `cdk-package`, but from a quick look, it seems that it supports many targets at once (essentially calling `pacmak` the same way as in lines 47-49). So instead of creating `pre-only` and `post-only` options, wouldn't it be simpler to call `cdk-package` instead of `pacmak` directly, and let it run the pre and post steps normally?

In other words, replacing these three separate steps with:

```bash
npx lerna run --scope aws-cdk-lib package -- --verbose --targets $(cat $TMPDIR/jsii.txt)
```

Reply:
I've implemented and tested both approaches, and I think we should stick with my current approach, even though it is more complex in the scripting code. The two major downsides to using lerna are:
1. 5 minutes is added to the pack time
2. Modifications to the nx.json config are required. And the change forces us to pack ALL dependencies of the packages that need to be packed, including packages in `tools` that don't need to be packed. 

I've made the change and then reverted it so you can take a look if you want: https://github.com/aws/aws-cdk/pull/25239/commits/b50c981a68cab989e249302ecdc68857e55a2f34